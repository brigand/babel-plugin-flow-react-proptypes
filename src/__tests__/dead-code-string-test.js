const babel = require('babel-core');
const { minify } = require('uglify-es');

const content = `
var React = require('react');

type Props = { x: string };
const Foo = (props: Props) => <div />;
`;

const getOpts = opts => ({
  babelrc: false,
  presets: [['es2015', { modules: false }], 'stage-1', 'react'],
  plugins: ['syntax-flow', [require('../'), opts]],
});

it('dead-code-string', () => {
  const res = babel.transform(content, getOpts({ deadCode: '__PROD__' })).code;
  expect(res).toMatch(/__PROD__\s*\?/);
  expect(res).toMatchSnapshot();
});

it('dead-code-string uglify', () => {
  const res = babel.transform(content, getOpts({ deadCode: '__PROD__' })).code
    .replace(/__PROD__/, 'true');
  const { code: min } = minify(res, { toplevel: true });
  expect(min).not.toMatch(/prop-types/);
});

it('dead-code-string with esm', () => {
  const res = babel.transform(content, getOpts({ deadCode: '__PROD__', useESModules: true })).code;
  expect(res).toMatch(/__PROD__\s*\?/);
  expect(res).toMatchSnapshot();
});

it('dead-code-string uglify with esm', () => {
  const res = babel.transform(content, getOpts({ deadCode: '__PROD__', useESModules: true })).code
    .replace(/__PROD__/, 'true');
  const { code: min } = minify(res, { toplevel: true });
  expect(min).not.toMatch(/string/);
  expect(min).toMatchSnapshot();
});
