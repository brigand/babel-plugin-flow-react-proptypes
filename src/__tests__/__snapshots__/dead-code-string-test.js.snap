// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dead-code-string 1`] = `
"'use strict';

var React = require('react');

var Foo = function Foo(props) {
  return React.createElement('div', null);
};
Foo.propTypes = __PROD__ ? null : {
  x: require('prop-types').string.isRequired
};"
`;
